### ISSUE TITLE
[Question] Discrepancy in reproducing Zero-shot baseline results

### ISSUE THREAD

**Issue by Suzukitakeko** [AuroraZengfh/DIR-Merging#7]
Hi authors,

Thank you for your excellent work and for open-sourcing the code!

I'm currently trying to reproduce the results from Table 2 in your paper. While doing so, I've encountered some noticeable discrepancies specifically when reproducing the **Zero-shot** baseline.

**Notably, when I use your provided LoRA models and merge them using the `RobustMerge` method, my results are very close to what you reported in the paper.** This leads me to believe that my overall environment and data setup are likely correct, and the issue might be specific to the details of the Zero-shot evaluation.

Here is a comparison between the paper's Zero-shot results and my own:

| Task | Paper's Result (Table 2) | My Reproduced Result | Diff. |
| :--- | :---: | :---: | :---: |
| **Seen Tasks** | | | |
| SciQA | 61.73 | IMG-Accuracy: 67.87% | +6.14 |
| ImageNet | 40.87 | 50.26% | +9.39 |
| VQAv2 | 62.88 | 67.49% | +4.61 |
| REC | 36.10 | 36.35% | +0.25 |
| OCR | 41.16 | 64.23% | +23.07 |
| VizWiz | 41.03 | 39.57% | -1.46 |
| Flickr | 49.07 | 41.19% | -7.88 |
| IconQA | 14.09 | 18.93% | +4.84 |
| **Unseen Tasks**| | | |
| AVQA | 51.62 | 70.04% | +18.42 |
| Image-R | 28.27 | 40.84% | +12.57 |
| S2W | 5.98 | 7.68% | +1.70 |
| TabMWP | 15.01 | 16.77% | +1.76 |

**My Evaluation Script Example:**

To help you identify the potential issue, here is the script I used to run the Zero-shot evaluation.

```python
# ===============================================================
#!/bin/bash

CHUNKS=1
IDX=0

MODELPATH=$1
GPU=$2

RESULT_DIR="./results/ScienceQA"
echo $RESULT_DIR

CUDA_VISIBLE_DEVICES=$GPU python -m llava.eval.model_vqa_science \
    --model-path $MODELPATH \
    --question-file /RobustMerge/MM-MergeBench/Seen_data/ScienceQA/test.json \
    --image-folder /RobustMerge/datasets \
    --answers-file $RESULT_DIR/${CHUNKS}_${IDX}.jsonl \
    --num-chunks $CHUNKS \
    --chunk-idx $IDX \
    --temperature 0 \
    --conv-mode vicuna_v1 &

wait

output_file=$RESULT_DIR/merge.jsonl

# Clear out the output file if it exists.
> "$output_file"

# Loop through the indices and concatenate each file.
for IDX in $(seq 0 $((CHUNKS-1))); do
    cat $RESULT_DIR/${CHUNKS}_${IDX}.jsonl >> "$output_file"
done

python /RobustMerge/llava/eval/eval_science_qa.py \
    --base-dir /RobustMerge/datasets/ScienceQA/ \
    --result-file $output_file \
    --output-file $RESULT_DIR/output.jsonl \
    --output-result $RESULT_DIR/output_result.jsonl 
# ===============================================================
```

I was wondering if there is a specific pre-trained checkpoint that should be used for the Zero-shot baseline (perhaps different from the base model used for merging)? Or are there any specific arguments or configurations for the Zero-shot evaluation that I might have missed?

Any clarification you can provide would be greatly appreciated. Thanks again

**Comment by Suzukitakeko** [comment]
Hi, thank you so much for your prompt reply!

I can confirm that I have been using `Llava-1.5-7b` as the base model for the zero-shot evaluation.

Interestingly, while the zero-shot results still differ, I went a step further and tested the performance for the **Individual** and **RobustMerge** methods using the LoRA checkpoints you provided. I found that my reproduced results for these two methods align almost perfectly with the figures in your paper.

Here is the updated comparison table with my reproduced results:

| Method | SciQA | Image | VQA | REC | OCR | VizWiz | Flickr | IconQA | AVQA | Image-R | S2W | TabMWP |
| :--- | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: |
| **Individual (Paper)** | 83.74 | 96.02 | 67.58 | 43.40 | 65.50 | 64.80 | 57.29 | 75.54 | - | - | - | - |
| **Individual (Mine)** | **83.64** | **96.00** | **67.62** | **43.38** | **64.97** | **64.83** | **57.24** | **75.51** | - | - | - | - |
| **Zero-Shot (Paper)** | 61.73 | 40.87 | 62.88 | 36.10 | 41.16 | 41.03 | 49.07 | 14.09 | 51.62 | 28.27 | 5.98 | 15.01 |
| **Zero-Shot (Mine)** | **67.87** | **50.26** | **67.49** | **36.35** | **64.23** | **39.57** | **41.19** | **18.93** | **70.04** | **40.84** | **7.68** | **16.77** |
| **RobustMerge (Paper)** | 73.43 | 65.54 | 67.20 | 44.80 | 62.97 | 46.61 | 52.80 | 45.90 | 79.30 | 45.79 | 9.23 | 17.62 |
| **RobustMerge (Mine)** | **73.53** | **65.56** | **67.18** | **44.78** | **61.92** | **45.58** | **52.83** | **45.91** | **79.39** | **45.74** | **9.27** | **17.71** |

As you can see, the `Individual` and `RobustMerge` results are an almost perfect match. This strongly suggests that my environment, datasets, and base evaluation pipeline are correct. The discrepancy, therefore, seems to be specific to the zero-shot evaluation process.

Since my setup for the merging tasks appears to be correct, I'd like to take this opportunity to ask a quick question about the experimental design for the **Multi-Task** baseline. Am I correct in assuming that this is trained by **combining all the training data from the "seen tasks" in MM-MergeBench and then LoRA fine-tuning the base model for one epoch**?

Thank you again for your time and clarification

**Comment by AuroraZengfh** [comment]
That's really weird. I cannot come up with an idea for this. What GPU do you test on? We conduct on A6000 and I'm not sure if this would cause such difference and more importantly, **only on zero-shot**. Maybe you can fine-tune individual models by yourself to see if the individual models get a much higher performance.

Regarding Multi-task baseline, your understanding in consistent with the setting conducted in the paper.

**Comment by AuroraZengfh** [comment]
Great appreciate your diligent work! We also do encounter some differences during reproduction. Since it does not affect the performance on individual and our method, feel free to use the results reproduced on your own in your paper.
